e2e/stress/DeviceTelemetryTest/DeviceTelemetryTest.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    public class DeviceTelemetryTest
+    {
+        int _messageSize;
+        int _messageCount;
+        int _parallelCount;
+        TransportType _protocol;
+        ResultWriter _writer;
+        DeviceClient _client;
```

Comment:
`_client` could be local variable in `RunTestAsync`

Reply:
In C# is preferred to init in declaration. In dotnet/corefx we were following something similar to this: https://stackoverflow.com/questions/24551/initialize-class-fields-in-constructor-or-at-declaration (can't find the doc on dotnet/corefx).

Reply:
```
public async Task RunTestAsync()
{
    DeviceClient deviceClient = CreateDevice();
    ...
    await deviceClient.OpenAsync(timeout).ConfigAwait(false);
    ...
    for (int i = 0; i < _parallelCount; i++)
    {
        tasks.Add(SendMessageAsync(deviceClient));
    }
    ....
```

Reply:
I don't understand why I would need to pass in the deviceClient as a parameter to member functions when things are encapsulated in a single class.